app/assets/javascripts/angular/controllers/pages/create_recipe_controller.js.erb

```diff
@@ -1,43 +1,70 @@
+  $scope.measurements = ["pod(s)","container(s)","packet(s)", "ounce(s)", "sheet(s)", "basket(s)", "slab(s)","dash(es)","milliliter(s)","x","shell(s)","rack(s)","quart(s)","bunch(es)","serving(s)","chunk(s)","clove(s)","liter(s)","twist(s)","gallon(s)","grind(s)","stalk(s)","tub(s)","piece(s)","pint(s)","feet","prune(s)","fluid","shot(s)","envelope(s)","gram(s)","bottle(s)","slice(s)","inch(es)","pound(s)","oz","count(s)","tray(s)","kilogram(s)","jar(s)","leaf(s)","head(s)","round(s)","whole","squeeze(s)","wedge(s)","pat(s)","spash(es)","bone(s)","pounch(es)","hank(s)","loaf(s)","large","bag(s)","spear(s)","pinch(es)","strip(s)","thin","roll(s)","medium","jumbo","bite","square(s)","cube(s)","rib(s)","sprig(s)","drop(s)","crown(s)","cup(s)","leg(s)","stick(s)","tablespoon(s)","scoop(s)","link(s)","package(s)","mini","sleeve(s)","ear(s)","bulb(s)","box(es)"].sort();
 
-  var hourArray = [];
-  for (var i = 0; i < 24 ; i++) {
-    hourArray.push(i);
-  }
+  $scope.create_parameters = {};
 
-  var minuteArray = [];
-  for (var i = 0; i < 60 ; i++) {
-    minuteArray.push(i);
-  }
+  // var user = UserServices;
+  // $scope.create_parameters.author = UserServices.username;
 
-  $scope.hours = hourArray;
-  $scope.minutes = minuteArray;
+  $scope.create_parameters.title = '';
 
-  $scope.measurements = measures.sort()
-  // console.log($scope.measurements);
+  $scope.create_parameters.description = '';
 
-  var ingredients = [];
-  $scope.addIngredient = function() {
-    console.log($scope.addIngredNumber,$scope.addIngredUnit,$scope.addIngredName);
-
-    ingredient["number"] = $scope.addIngredNumber;
-    ingredient["unit"] = $scope.addIngredUnit;
-    ingredient["ingredient"] = $scope.addIngredName;
-    ingredients.push(ingredient);
-    $('.added-ingredients tbody').prepend("<tr class='added-ingredient'><td class='col-xs-2'></td><td class='col-xs-1'>-</td><td class='col-xs-1'>"+$scope.addIngredNumber+"</td><td class='col-xs-2'>"+$scope.addIngredUnit+"</td><td class='col-xs-6'>"+$scope.addIngredName+"</td></tr>");
-  }
+  { // add ingredients function
+    $scope.create_parameters.all_ingredients = [];
 
-  // time slider function
-  {
-    $scope.search_parameters.timeframe = [0,120];
+    var checkDuplicates = function (keyword,array) {
+      for (var i= 0 ; i < array.length ; i++) {
+        if (keyword == array[i]) {
+          return true;
+        }
+      }
+    };
+
+    $scope.newIngredient = function() {
```

Comment:
Try to use verbs for functions as they do something, unlike nouns. $scope.addIngredient() for instance?